src/com/oop1/entity/Inventory.java

```
@@ -17,104 +24,76 @@
      * modifications to the inventory should be performed
      * via the other methods on this class.
      */
-    public Collection<Item> getAllItems() {
-        return Collections.unmodifiableCollection(items);
-    }
+    public Collection<TakeableItem> getAllItems() { return Collections.unmodifiableCollection(items); }
 
-    public Collection<Item> getEquippedItems() {
-        // TODO: implement this
-        return Collections.emptyList();
-    }
+    /**
+     * Get all the equipped items
+     *
+     * @return an unmodifiable collection of the items in the inventory.
+     * modifications to the inventory should be performed
+     * via the other methods on this class.
+     */
+
+    public Collection<TakeableItem> getEquippedItems() { return Collections.unmodifiableCollection(equipment); }
 
-    public void equipItem(Item item) {
-        // TODO: implement this
+    public void equipItem(TakeableItem item) {
+        if (items.contains(item)) {
+            items.remove(item);
+            equipment.add(item);
+        }
+    }
 
+    public void unequipItem(TakeableItem item) {
+        if (equipment.contains(item)) {
```

we should probably throw an error if we try to unequip an item that's not equipped. same for equipping an item not in the inventory in **equipItem()**